mitodl/mitxpro
# feat: Use poetry instead of requirements files

### What are the relevant tickets?
https://github.com/mitodl/mitxpro/issues/2687

### Description (What does it do?)
This change involves using `poetry` for Python dependency management.

### How can this be tested?
Here's how to make sure it's working correctly:
- Make sure the tests pass when running them on CI.
- Check if everything is functioning locally as expected by running the command `docker compose up --build`
- For testing on Heroku, it's best to wait for the RC.
- Attempt to include a new package. You will have to use the poetry command-line tool to add or remove the package. This tool will automatically make the necessary adjustments to the `pyproject.toml` and `poetry.lock` files.
  - If you have poetry installed on your host: 
    - Use `poetry add/remove --lock <package-name>` for adding or removing a library. Note that adding a library might also install it on your host If you omit `--lock`.
  - If you don't have poetry on your computer, but you're using Docker:
    - Run `docker compose exec -u root web poetry add/remove <package-name>` for adding or removing a library. Keep in mind that for changes to persist between containers, you'll need to rebuild them.
